### ISSUE TITLE
ESLint falls into infinite loop if used with css-loader in webpack

### ISSUE THREAD

**Issue by wyvernnot** [eslint/eslint#6962]
**What version of ESLint are you using?**

```
2.7.0
````

**What parser (default, Babel-ESLint, etc.) are you using?**

```
babel-eslint
```

**Please show your full configuration:**

It is a React project, built with webpack, configured with following loaders

```
babel-loader
eslint-loader
css-loader
file-loader
less-loader
style-loader
```

We use `eslint-config-airbnb` default settings, which will enable the `no-multi-spaces` check bu default.

```js
// disallow use of multiple spaces
   'no-multi-spaces': 2,
```

webpack module settings

```js
  module: {
    preLoaders: [
      {
        test: /\.(js|es6|jsx)$/,
        loader: 'eslint-loader',
        exclude: /node_modules/
      }
    ],
    loaders: [
      {
        test: /\.(jsx|es6)$/,
        loader: 'babel'
      },
      {
        test: /\.less$/,
        loader: 'style!css!less'
      },
      {
        test: /\.css$/,
        loader: 'style!css'
      }
    ]
  },
```

**What did you do? Please include the actual source code causing the issue.**

I use the eslint-loader in my webpack development mode to check a file which has one space line in it

```js
'user strict'

function foo(){
  // bar
}
```

**What did you expect to happen?**

I was expecting eslint to check it quickly.

**What actually happened? Please include the actual, raw output from ESLint.**

The build actually hangs there.

**Comment by wyvernnot** [comment]
I found the root cause.

In `no-multiple-empty-lines.js`, there is code like this:

```js
 currentLocation = trimmedLines.indexOf("", currentLocation + 1);
```

This line of code actually assumes the indexOf is native or according to [standard](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/indexOf
):

```
arr.indexOf(searchElement[, fromIndex = 0])
```

In rare case, the `indexOf` will be polyfilled in a non-standard way by another module in the project.

```js
if(!Array.indexOf)
  Array.prototype.indexOf = function (vItem) {
    for (var i=0; i < this.length; i++) {
      if (vItem === this[i]) {
        return i;
      }
    }
    return -1;
  };
}
```

[above code direct link](https://github.com/redhivesoftware/math-expression-evaluator/blob/d6a82d8225232bbcb33246dc2337ca3a524adc99/src/postfix.js
)


This becomes a disaster, because following code enters infinite loop as the `fromIndex` parameter is completely ignored.

```
 currentLocation = trimmedLines.indexOf("", currentLocation + 1);
```

**Comment by gyandeeps** [comment]
I am personally not in favor of introducing such changes because some code has polyfilled some core functionality in a non-standard (not according to the specs) way. By doing this we will say, we are open to make changes on ESlint juts because your code uses non-standard javascript function implementations.

This is my thought, I will wait to see what the team has to say on this.

**Comment by platinumazure** [comment]
I agree with @gyandeeps. Relevant passage from [The Codeless Code](http://thecodelesscode.com/case/154):

> “When the force of gravity suddenly changes direction, or the gods decree that wood shall become dust and rope shall weigh as much as lead, how do you keep travellers from plunging into the abyss?”
> 
> The builder said: “We simply do not plan for such absurd possibilities.”

I'm not sure how we can plan for this, either.

**Comment by kaicataldo** [comment]
@wyvernnot Want to make it clear that we appreciate the bug report - thanks for contributing! 

I'm in agreement with the first two opinions voiced here - I don't see how we can account for all non-standard polyfills, and I think trying to do so sets a dangerous precedent.

**Comment by ilyavolodin** [comment]
I also agree. JavaScript is a very flexible language. Anything can be overwritten by the user at any given point. There's no way to predict and protect against such actions. Thank you for suggestions, but there seems to be a strong agreement between the members of the team, that this is not something we should try to protect against.
